utils/generic-components.stanza

```diff
@@ -585,12 +585,18 @@ public defn gen-led-cmp (pkg-name:String) :
 ; ========================================================
 ; These functions attach a passive component between two pins. e.g. strap a pull up resistor between an IO and VDD
 
-public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, capacitance:Double|Toleranced):
+public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, params:Tuple<KeyValue>):
   inside pcb-module:
-    val cap = cap-strap(gnd-pin, power-pin, capacitance)
+    val cap = cap-strap(gnd-pin, power-pin, params)
     short-trace(cap.p[2], power-pin)
     cap
 
+public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, capacitance:Double|Toleranced, tol:Double):
+  bypass-cap-strap(power-pin, gnd-pin, ["capacitance" => capacitance, "tolerance" => tol])
+
+public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, capacitance:Double|Toleranced):
+  bypass-cap-strap(power-pin, gnd-pin, ["capacitance" => capacitance])
+
```

Comment:
So this might be a good place to use the undocumented optional/named arguments in Stanza.

The function could look something like (no guarantees that this code works 😸 ):
```
public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, -- params:Tuple<KeyValue> = false capacitance:Double|Toleranced|False = false, tol:Double|False = false)
  var my-params:Tuple<KeyValue>
  if params is-not False:
    add(my-params, params)
  if capacitance is-not False :
    add(my-params, ["capacitance" => capacitance])
  if tol is-not False:
    add(my-params, ["tolerance" => tol])
    inside pcb-module:
  val cap = cap-strap(gnd-pin, power-pin, params)
  short-trace(cap.p[2], power-pin)
  cap
```

Reply:
You would then need to call the function as
`bypass-cap-strap(pina, pinb, params=params-to-send)`
`bypass-cap-strap(pina, pinb, capacitance=1.0e-6)` 
and so on...

Reply:
Maybe as a way to not break the existing code - we could consider:

```
public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, capacitance:Double|Toleranced, tol:Double -- params:Tuple<KeyValue> = false):
  .. Handle the `params` arg here. 
  bypass-cap-strap(power-pin, gnd-pin, ["capacitance" => capacitance, "tolerance" => tol])

public defn bypass-cap-strap (power-pin:JITXObject, gnd-pin:JITXObject, capacitance:Double|Toleranced -- params:Tuple<KeyValue> = false):
  .. Handle the `params` arg here
  bypass-cap-strap(power-pin, gnd-pin, ["capacitance" => capacitance])
```

That would add the ability to include the parameters while keeping the positional argument versions intact. 

I think we need a plan for how we are going to update this interface. Just doing it adhoc PR to PR doesn't seem like a good idea. Better to rip the bandaid off and a have a plan for supporting legacy implementations.

Reply:
I agree with the concern to not upend existing usage. But unfortunately the only way that we are going to use the full power of the named and optional arguments is to rip the band-aid off.